test/services/service_status.service.test.js

```diff
@@ -3,38 +3,176 @@
 // Test framework dependencies
 const Lab = require('@hapi/lab')
 const Code = require('@hapi/code')
+const Sinon = require('sinon')
+const Nock = require('nock')
+const Proxyquire = require('proxyquire')
 
-const { describe, it } = exports.lab = Lab.script()
+const { describe, it, beforeEach, afterEach } = exports.lab = Lab.script()
 const { expect } = Code
 
 // Thing under test
-const ServiceStatusService = require('../../app/services/service_status.service.js')
-
-const firstRow = [
-  { text: 'Cell 1.1' },
-  { text: 'Cell 1.2' },
-  { text: 'Cell 1.3' },
-  { text: 'Cell 1.4' },
-  { text: 'Cell 1.5' }
-]
-
-const secondRow = [
-  { text: 'Cell 2.1' },
-  { text: 'Cell 2.2' },
-  { text: 'Cell 2.3' },
-  { text: 'Cell 2.4' },
-  { text: 'Cell 2.5' }
-]
+// Normally we'd set this to `= require('../../app/services/service_status.service')`. But to control how
+// `child_process.exec()` behaves in the service, after it's been promisfied we have to use proxyquire.
+let ServiceStatusService // = require('../../app/services/service_status.service')
 
 describe('Service Status service', () => {
-  describe('importRows data', () => {
-    it('returns data in the expected format', async () => {
-      const { importRows } = await ServiceStatusService.go()
+  beforeEach(() => {
+    // These requests will remain unchanged throughout the tests. We do alter the ones to the AddressFace and the
+    // water-api (foreground-service) though, which is why they are defined separately in each test.
+    Nock('http://localhost:8020')
+      .get('/status')
+      .reply(200, { status: 'alive' },
+        [
+          'x-cma-git-commit', '273604040a47e0977b0579a0fef0f09726d95e39',
+          'x-cma-docker-tag', 'ghcr.io/defra/sroc-charging-module-api:v9.99.9'
+        ]
+      )
 
-      expect(importRows).to.equal([
-        firstRow,
-        secondRow
+    Nock('http://localhost:8012').get('/health/info').reply(200, { version: '8.0.12', commit: '83d0e8c' })
+    Nock('http://localhost:8011').get('/health/info').reply(200, { version: '8.0.11', commit: 'a7030dc' })
+    Nock('http://localhost:8007').get('/health/info').reply(200, { version: '8.0.7', commit: 'a181fb1' })
+    Nock('http://localhost:8002').get('/health/info').reply(200, { version: '8.0.2', commit: '58bd0c1' })
+    Nock('http://localhost:8000').get('/health/info').reply(200, { version: '8.0.0', commit: 'f154e3f' })
+    Nock('http://localhost:8008').get('/health/info').reply(200, { version: '8.0.8', commit: 'f154e3f' })
+    Nock('http://localhost:8003').get('/health/info').reply(200, { version: '8.0.3', commit: '2ddff3c' })
+    Nock('http://localhost:8004').get('/health/info').reply(200, { version: '8.0.4', commit: '09d5261' })
+    Nock('http://localhost:8006').get('/health/info').reply(200, { version: '8.0.6', commit: 'b181625' })
```

Comment:
Do we want to hardcode these urls or would we prefer to read them in from our config?

Reply:
🤷 We hardcode the values in our `.github/ci.yml` so the result would be the same. I guess there is an argument that you'd only have to declare the URL once. 🤷 

That said this is in a unit test. And IMHO I prefer to be very verbose and explicit about any test setup so it's clear what are the parameters of the test.

But honestly, this is not a hill I'm gonna die on so let me know your preference.

Reply:
Happy to leave it as-is :grin: